platforms/karaf/features/src/main/resources/feature.xml

```diff
@@ -59,6 +59,7 @@
     <bundle>mvn:io.fabric8/kubernetes-model-scheduling/${project.version}</bundle>
     <bundle>mvn:io.fabric8/kubernetes-model-storageclass/${project.version}</bundle>
     <bundle>mvn:io.fabric8/openshift-model/${project.version}</bundle>
+    <bundle>mvn:io.fabric8/kubernetes-httpclient-okhttp/${project.version}/jar/bundle</bundle>
```

Comment:
This is tricky, so consider it more like thinking out loud rather than a change request.

I would move the `kubernetes-httpclient-okhttp` to it's own feature, along with it's dependencies. I would also provide a feature for `kubernetes-httpclient-okhttp`.

Then I would either decouple `kubernetes-client` from `kubernetes-httpclient-okhttp` and let users decide what to use or just refer to it (but most probably the first). The former adds an extra step for the user, the latter is less flexible. In both cases the idea is that we communicate this way which are the pluggable parts.

Again, this is something that we can do afterwards. I prefer merging this faster and add this touches later on.

Reply:
Opening as #3878